uwrt_mars_rover_drivetrain/uwrt_mars_rover_drivetrain_hw/drivers/roboteq_driver/src/CanopenInterface.cpp

```diff
@@ -97,8 +97,10 @@ CanopenInterface::CanopenInterface(canid_t roboteq_can_id, const std::string& if
   can_receive_filter[1].can_id = SDO_RESPONSE_COB_ID_OFFSET + roboteq::CanopenInterface::roboteq_can_id_;
   can_receive_filter[1].can_mask = (CAN_EFF_FLAG | CAN_RTR_FLAG | CAN_EFF_MASK);
 
-  //   struct timeval receive_timeout = {.tv_usec=500000}; //0.5 seconds
-  // setsockopt(socket_handle_, SOL_SOCKET, SO_RCVTIMEO, &receive_timeout, sizeof(receive_timeout));
+  constexpr unsigned RECEIVE_TIMEOUT_US{500000};  // 0.5 seconds
+  struct timeval receive_timeout = {0};
+  receive_timeout.tv_usec = RECEIVE_TIMEOUT_US;
+  setsockopt(socket_handle_, SOL_SOCKET, SO_RCVTIMEO, &receive_timeout, sizeof(receive_timeout));
```

Comment:
Required since we cant enable c++20 yet cuz its not bundled in default cmake installation in 18.04

Reply:
kk. ig we can still update g++ but its not woth the effort?

Reply:
also do we actually compile this stuff on the jetson? at my coop, we cross-compile locally, then copy all shared libs and the executables over to the jetson

Reply:
It's easier to stick with the default version of compilers available on the platform that the ROS version is tied too. that way we run into less issues with other packages we compile from src

Reply:
And yes we do compile on the jetson. we have a jetson xavier AGX, which i think is actually better than most members laptops(32GB RAM and 8-core ARM v8 64-bit CPU w/ 8MB L2 + 4MB L3). Here's some benchmarks against a "high end" laptop:

![image](https://user-images.githubusercontent.com/16629657/113465229-6c21d700-9400-11eb-9054-c5bd817e28b9.png)

Cuda core performance(not relevant for compile)
![image](https://user-images.githubusercontent.com/16629657/113465237-8360c480-9400-11eb-8c25-3dc463dec2a9.png)

My personal desktop with like 24 threads and 32gb ram would definitely be faster to cross compile on but its just not worth it to setup for just me lol.